src/ngx_http_lua_socket_tcp.c

```diff
@@ -4908,65 +4901,30 @@ ngx_http_lua_socket_push_input_data(ngx_http_request_t *r,
     nbufs = 0;
     ll = NULL;
 
+    luaL_buffinit(L, &luabuf);
     for (cl = u->bufs_in; cl; cl = cl->next) {
+        ptrdiff_t chunk_size;
```

Comment:
hey @agentzh, IMHO, type choosing and variable scoping is not a matter of coding style. Honestly, I don't see how it is better to store the difference of two pointers in something else than `ptrdiff_t` (specially, using an unsigned type). Also, I don't see see how declaring a variable in the begin of a block is inconsistent with nginx style. Do you mean that we can use only the ANSI C subset that was previously used on the nginx core? (Sorry, but that doesn't make much sense to me =(